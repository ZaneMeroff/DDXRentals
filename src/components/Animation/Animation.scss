$randomNumber1: random(500);
$randomNumber2: random(500);
// $randomNumber3: random(500);

.fade-in {
	 animation: fade-in 1.2s linear infinite alternate both;
	 height: 20px;
	 width: auto;
	 z-index: 0;

}

#snowflake {
	left: $randomNumber2 + px;
	position: absolute;
	top: $randomNumber2 + px;
}

@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

.snowflake-container {
	height: 100vh;
	overflow: hidden;
	position: fixed;
	top: 0;
	left: 0;
	width: 100%;
}
