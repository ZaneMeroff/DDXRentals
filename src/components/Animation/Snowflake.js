import React from 'react';
import snowflake from '../../data/snowflake.png';

const Snowflake = (props) => {
  console.log(props.locationY);
  return (
    <img src={ snowflake } className='fade-in' style={{position:'absolute', top:props.locationX, left:props.locationY + 100}} alt='snowflake animation'/>
  )
}

export default Snowflake;
