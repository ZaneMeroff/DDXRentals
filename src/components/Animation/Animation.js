import React from 'react';
import './Animation.scss';
import Snowflake from './Snowflake.js';

const Animation = (props) => {
  let snowflakeAmt = 150;
  let snowflake = [];
  if (props.amount.value === 'Business') {
    snowflakeAmt = 100;
  } else if(props.amount.value === 'Recreational') {
    snowflakeAmt = 300;
  } else if(props.amount.value === 'EXTREME') {
    snowflakeAmt = 1000;
  }
  for (var i = 0; i < `${snowflakeAmt}`; i ++) {
    snowflake.push(Math.floor(Math.random() * 1000))
  }
  const flakes = snowflake.map(flake => <Snowflake locationX={flake - 50} locationY={Math.floor(Math.random() * 1500) - 100} key={Math.random()}/>);
  return (
    <div>
      {flakes}
    </div>
  )
}

export default Animation;
