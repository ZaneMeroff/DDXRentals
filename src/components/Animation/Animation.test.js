import React from 'react';
import ReactDOM from 'react-dom';
import Animation from './Animation';
import { shallow } from 'enzyme';
import { mockRandom } from 'jest-mock-random';

describe('Animation', () => {

  beforeEach(() => {
    mockRandom([0.1, 0.2, 0.3, 0.6]);
 });

  it('should be an instance of Animation', () => {
    const wrapper = shallow(<Animation />);
    expect(wrapper).toMatchSnapshot();
  });

});
