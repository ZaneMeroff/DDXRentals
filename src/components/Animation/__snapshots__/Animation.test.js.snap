// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animation should be an instance of Animation 1`] = `
<div>
  <Snowflake
    key="0.1421407308947824"
    locationX={755}
    locationY={833}
  />
  <Snowflake
    key="0.6297970655364975"
    locationX={368}
    locationY={319}
  />
  <Snowflake
    key="0.10526963350024254"
    locationX={40}
    locationY={66}
  />
  <Snowflake
    key="0.01742117974340851"
    locationX={292}
    locationY={633}
  />
  <Snowflake
    key="0.7413275648605562"
    locationX={328}
    locationY={1399}
  />
  <Snowflake
    key="0.007514296462558523"
    locationX={35}
    locationY={1130}
  />
  <Snowflake
    key="0.8859732740469233"
    locationX={606}
    locationY={924}
  />
  <Snowflake
    key="0.8944114944657646"
    locationX={908}
    locationY={637}
  />
  <Snowflake
    key="0.1150338239401052"
    locationX={275}
    locationY={553}
  />
  <Snowflake
    key="0.46561624853770955"
    locationX={619}
    locationY={408}
  />
  <Snowflake
    key="0.40167006937437333"
    locationX={932}
    locationY={1220}
  />
  <Snowflake
    key="0.15434521160893167"
    locationX={342}
    locationY={347}
  />
  <Snowflake
    key="0.16822467757641313"
    locationX={593}
    locationY={1387}
  />
  <Snowflake
    key="0.6963791364857628"
    locationX={640}
    locationY={7}
  />
  <Snowflake
    key="0.1341573570448924"
    locationX={501}
    locationY={1374}
  />
  <Snowflake
    key="0.023122569659731695"
    locationX={606}
    locationY={67}
  />
  <Snowflake
    key="0.7149333567678735"
    locationX={298}
    locationY={187}
  />
  <Snowflake
    key="0.7839848940923035"
    locationX={657}
    locationY={819}
  />
  <Snowflake
    key="0.5515451384766998"
    locationX={393}
    locationY={1051}
  />
  <Snowflake
    key="0.9236091410722815"
    locationX={329}
    locationY={602}
  />
  <Snowflake
    key="0.7276816205850603"
    locationX={721}
    locationY={814}
  />
  <Snowflake
    key="0.6985113973771129"
    locationX={2}
    locationY={394}
  />
  <Snowflake
    key="0.9016029108722188"
    locationX={714}
    locationY={281}
  />
  <Snowflake
    key="0.21494506278853365"
    locationX={221}
    locationY={83}
  />
  <Snowflake
    key="0.19621964016240145"
    locationX={688}
    locationY={946}
  />
  <Snowflake
    key="0.15839728258684116"
    locationX={355}
    locationY={1010}
  />
  <Snowflake
    key="0.22142143667096148"
    locationX={875}
    locationY={1047}
  />
  <Snowflake
    key="0.5318999335903378"
    locationX={450}
    locationY={1085}
  />
  <Snowflake
    key="0.4211306554963732"
    locationX={728}
    locationY={185}
  />
  <Snowflake
    key="0.9893578516458452"
    locationX={191}
    locationY={565}
  />
  <Snowflake
    key="0.4169639194257284"
    locationX={851}
    locationY={429}
  />
  <Snowflake
    key="0.9901856978835271"
    locationX={739}
    locationY={360}
  />
  <Snowflake
    key="0.3310625930544926"
    locationX={385}
    locationY={1457}
  />
  <Snowflake
    key="0.31322984854048097"
    locationX={930}
    locationY={639}
  />
  <Snowflake
    key="0.26060378242453375"
    locationX={337}
    locationY={1381}
  />
  <Snowflake
    key="0.13542713596301925"
    locationX={929}
    locationY={421}
  />
  <Snowflake
    key="0.4774092738127915"
    locationX={648}
    locationY={143}
  />
  <Snowflake
    key="0.6662812296041689"
    locationX={78}
    locationY={1295}
  />
  <Snowflake
    key="0.5273256940020408"
    locationX={577}
    locationY={969}
  />
  <Snowflake
    key="0.31466306471360994"
    locationX={197}
    locationY={1197}
  />
  <Snowflake
    key="0.667082871140219"
    locationX={381}
    locationY={468}
  />
  <Snowflake
    key="0.43698677047481116"
    locationX={884}
    locationY={673}
  />
  <Snowflake
    key="0.3640715211800547"
    locationX={888}
    locationY={682}
  />
  <Snowflake
    key="0.030886514250096253"
    locationX={435}
    locationY={788}
  />
  <Snowflake
    key="0.05307355111337109"
    locationX={210}
    locationY={359}
  />
  <Snowflake
    key="0.6972805666625095"
    locationX={249}
    locationY={1370}
  />
  <Snowflake
    key="0.25590467565136654"
    locationX={794}
    locationY={1414}
  />
  <Snowflake
    key="0.6946407783579376"
    locationX={762}
    locationY={1128}
  />
  <Snowflake
    key="0.6550038627798573"
    locationX={898}
    locationY={1047}
  />
  <Snowflake
    key="0.34445825137693187"
    locationX={154}
    locationY={1135}
  />
  <Snowflake
    key="0.5900772515181034"
    locationX={532}
    locationY={91}
  />
  <Snowflake
    key="0.7165780591808262"
    locationX={903}
    locationY={474}
  />
  <Snowflake
    key="0.8581469845195107"
    locationX={344}
    locationY={1148}
  />
  <Snowflake
    key="0.964098601284398"
    locationX={907}
    locationY={94}
  />
  <Snowflake
    key="0.42282756112833053"
    locationX={85}
    locationY={301}
  />
  <Snowflake
    key="0.703993460620211"
    locationX={868}
    locationY={1435}
  />
  <Snowflake
    key="0.2022162570500292"
    locationX={458}
    locationY={648}
  />
  <Snowflake
    key="0.21900477722814693"
    locationX={474}
    locationY={294}
  />
  <Snowflake
    key="0.48899463532150245"
    locationX={133}
    locationY={61}
  />
  <Snowflake
    key="0.1303015522536144"
    locationX={86}
    locationY={1457}
  />
  <Snowflake
    key="0.501956754973395"
    locationX={916}
    locationY={1233}
  />
  <Snowflake
    key="0.6670084826426659"
    locationX={679}
    locationY={463}
  />
  <Snowflake
    key="0.41227540599753265"
    locationX={503}
    locationY={44}
  />
  <Snowflake
    key="0.021410014995251858"
    locationX={73}
    locationY={503}
  />
  <Snowflake
    key="0.9165578347726693"
    locationX={410}
    locationY={827}
  />
  <Snowflake
    key="0.5154715054926955"
    locationX={594}
    locationY={72}
  />
  <Snowflake
    key="0.25677803004319966"
    locationX={19}
    locationY={1493}
  />
  <Snowflake
    key="0.4392545743732985"
    locationX={163}
    locationY={550}
  />
  <Snowflake
    key="0.7565308693086665"
    locationX={163}
    locationY={508}
  />
  <Snowflake
    key="0.3524386215890769"
    locationX={54}
    locationY={1379}
  />
  <Snowflake
    key="0.10809121521684784"
    locationX={810}
    locationY={871}
  />
  <Snowflake
    key="0.32610388715321803"
    locationX={228}
    locationY={1347}
  />
  <Snowflake
    key="0.8091543609872067"
    locationX={493}
    locationY={203}
  />
  <Snowflake
    key="0.8526213001435847"
    locationX={183}
    locationY={1210}
  />
  <Snowflake
    key="0.6989670511802077"
    locationX={483}
    locationY={1408}
  />
  <Snowflake
    key="0.1303172945101898"
    locationX={252}
    locationY={191}
  />
  <Snowflake
    key="0.059486499742704924"
    locationX={542}
    locationY={1143}
  />
  <Snowflake
    key="0.9857515948469378"
    locationX={423}
    locationY={407}
  />
  <Snowflake
    key="0.7449108784354093"
    locationX={398}
    locationY={142}
  />
  <Snowflake
    key="0.09617666553291215"
    locationX={494}
    locationY={1491}
  />
  <Snowflake
    key="0.5486855787006442"
    locationX={924}
    locationY={989}
  />
  <Snowflake
    key="0.9581716276582266"
    locationX={177}
    locationY={512}
  />
  <Snowflake
    key="0.6347965234656532"
    locationX={385}
    locationY={599}
  />
  <Snowflake
    key="0.5976270019848109"
    locationX={295}
    locationY={914}
  />
  <Snowflake
    key="0.713881944074803"
    locationX={415}
    locationY={1357}
  />
  <Snowflake
    key="0.30808069188810494"
    locationX={534}
    locationY={1218}
  />
  <Snowflake
    key="0.5182618839061641"
    locationX={850}
    locationY={19}
  />
  <Snowflake
    key="0.00629593640847248"
    locationX={70}
    locationY={945}
  />
  <Snowflake
    key="0.49875767863113296"
    locationX={279}
    locationY={1272}
  />
  <Snowflake
    key="0.09724165596783241"
    locationX={822}
    locationY={406}
  />
  <Snowflake
    key="0.6222433652394248"
    locationX={586}
    locationY={259}
  />
  <Snowflake
    key="0.7115719332085462"
    locationX={306}
    locationY={255}
  />
  <Snowflake
    key="0.6932223104698605"
    locationX={466}
    locationY={371}
  />
  <Snowflake
    key="0.13198845390632652"
    locationX={543}
    locationY={728}
  />
  <Snowflake
    key="0.7315872721916916"
    locationX={89}
    locationY={390}
  />
  <Snowflake
    key="0.30092257226334995"
    locationX={782}
    locationY={805}
  />
  <Snowflake
    key="0.8823894765492135"
    locationX={498}
    locationY={1130}
  />
  <Snowflake
    key="0.5684116126483691"
    locationX={617}
    locationY={393}
  />
  <Snowflake
    key="0.00024697196031464586"
    locationX={-13}
    locationY={478}
  />
  <Snowflake
    key="0.46627544138748545"
    locationX={229}
    locationY={858}
  />
  <Snowflake
    key="0.647815176993096"
    locationX={651}
    locationY={381}
  />
  <Snowflake
    key="0.9494758907654774"
    locationX={501}
    locationY={535}
  />
  <Snowflake
    key="0.5254039630315173"
    locationX={599}
    locationY={882}
  />
  <Snowflake
    key="0.4246129861686978"
    locationX={880}
    locationY={86}
  />
  <Snowflake
    key="0.39374335684436423"
    locationX={941}
    locationY={961}
  />
  <Snowflake
    key="0.8568117881640807"
    locationX={512}
    locationY={796}
  />
  <Snowflake
    key="0.43136189779999135"
    locationX={762}
    locationY={614}
  />
  <Snowflake
    key="0.24723865397552824"
    locationX={599}
    locationY={701}
  />
  <Snowflake
    key="0.8526125279156458"
    locationX={187}
    locationY={942}
  />
  <Snowflake
    key="0.1307042878625102"
    locationX={833}
    locationY={1428}
  />
  <Snowflake
    key="0.30004649936589467"
    locationX={775}
    locationY={156}
  />
  <Snowflake
    key="0.8061385599860214"
    locationX={347}
    locationY={1295}
  />
  <Snowflake
    key="0.08187853926157573"
    locationX={236}
    locationY={202}
  />
  <Snowflake
    key="0.6701184921014736"
    locationX={553}
    locationY={1356}
  />
  <Snowflake
    key="0.12084711591999087"
    locationX={389}
    locationY={938}
  />
  <Snowflake
    key="0.6086356430668483"
    locationX={291}
    locationY={505}
  />
  <Snowflake
    key="0.817331099611718"
    locationX={291}
    locationY={712}
  />
  <Snowflake
    key="0.6903812394016087"
    locationX={779}
    locationY={1447}
  />
  <Snowflake
    key="0.46057236462429807"
    locationX={611}
    locationY={1128}
  />
  <Snowflake
    key="0.4340963618972724"
    locationX={592}
    locationY={258}
  />
  <Snowflake
    key="0.8758540604635359"
    locationX={903}
    locationY={1406}
  />
  <Snowflake
    key="0.1804857763679364"
    locationX={101}
    locationY={957}
  />
  <Snowflake
    key="0.16993208464279963"
    locationX={563}
    locationY={941}
  />
  <Snowflake
    key="0.5395545610984882"
    locationX={691}
    locationY={120}
  />
  <Snowflake
    key="0.40449784930418375"
    locationX={200}
    locationY={50}
  />
  <Snowflake
    key="0.14721632684182517"
    locationX={192}
    locationY={730}
  />
  <Snowflake
    key="0.6330781273756678"
    locationX={226}
    locationY={1088}
  />
  <Snowflake
    key="0.4551668060477647"
    locationX={571}
    locationY={372}
  />
  <Snowflake
    key="0.6325341155914472"
    locationX={526}
    locationY={629}
  />
  <Snowflake
    key="0.30560648171758364"
    locationX={366}
    locationY={128}
  />
  <Snowflake
    key="0.5756314036700034"
    locationX={208}
    locationY={1404}
  />
  <Snowflake
    key="0.2240015435752747"
    locationX={51}
    locationY={808}
  />
  <Snowflake
    key="0.27388494371926586"
    locationX={692}
    locationY={101}
  />
  <Snowflake
    key="0.7709512943175201"
    locationX={622}
    locationY={1047}
  />
  <Snowflake
    key="0.6085660076927506"
    locationX={334}
    locationY={1410}
  />
  <Snowflake
    key="0.22088420209602022"
    locationX={451}
    locationY={558}
  />
  <Snowflake
    key="0.8379736172866488"
    locationX={858}
    locationY={1304}
  />
  <Snowflake
    key="0.5693420832125806"
    locationX={803}
    locationY={1186}
  />
  <Snowflake
    key="0.29309765384208775"
    locationX={117}
    locationY={797}
  />
  <Snowflake
    key="0.16856783810986342"
    locationX={671}
    locationY={1334}
  />
  <Snowflake
    key="0.26048947551254"
    locationX={272}
    locationY={660}
  />
  <Snowflake
    key="0.6845426705058595"
    locationX={390}
    locationY={1422}
  />
  <Snowflake
    key="0.5930973012972574"
    locationX={124}
    locationY={1481}
  />
  <Snowflake
    key="0.1860497266898251"
    locationX={628}
    locationY={760}
  />
  <Snowflake
    key="0.47009424550781764"
    locationX={884}
    locationY={86}
  />
  <Snowflake
    key="0.7745629840764923"
    locationX={321}
    locationY={754}
  />
  <Snowflake
    key="0.037656209456574885"
    locationX={503}
    locationY={1323}
  />
  <Snowflake
    key="0.16207050775820053"
    locationX={608}
    locationY={1253}
  />
  <Snowflake
    key="0.18848783759365983"
    locationX={-23}
    locationY={821}
  />
  <Snowflake
    key="0.9511614903085066"
    locationX={407}
    locationY={188}
  />
</div>
`;
